soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -238,6 +238,17 @@ static int nxp_lpc55xxx_init(const struct device *arg)
 	/* restore interrupt state */
 	irq_unlock(oldLevel);
 
+
+	/* Set the Submodule Clocks for FlexPWM */
+#if CONFIG_SOC_LPC55S36 && CONFIG_PWM
```

Comment:
I was being lazy, you probably want this to be:

#if defined(CONFIG_SOC_LPC55S36) && defined(CONFIG_PWM)